### ISSUE TITLE
.Net: Templatize Handlebars Plans

### ISSUE THREAD

**Issue by rithvikb23** [microsoft/semantic-kernel#5097]
I'm wondering if there is a way for developers to save handlebars plans as templates and execute a query against it.

Example: 
Query - What is the current price of apple?

Handlebars Plan:
{{!-- Step 1: Setting the Key Values --}} 
{{set "ticker" "AAPL"}}

{{!-- Step 2: Calling the Custom Helper --}}
{{set "response" ((CurrentPricePlugin-getCurrentPrice ticker=ticker}}

{{!-- Step 2: Output the Result --}}
{{json response}}

We would like it to be templatized as:

{{!-- Step 1: Setting the Key Values --}} 
{{set "ticker" $ticker"}}

{{!-- Step 2: Calling the Custom Helper --}}
{{set "response" ((CurrentPricePlugin-getCurrentPrice ticker=ticker}}

{{!-- Step 2: Output the Result --}}
{{json response}}

Where as you can see the ticker AAPL has been replaced by a variable. Now when a new query is ran like "What is the current price of NVDA?" The plan should just fill in the NVDA for $ticker.

Is there a current solution for this?

**Comment by Krzysztof318** [comment]
You can save generated plan via HandlebarsPlan.ToString() and create new instance of HandlebarsPlan(string plan).

**Comment by teresaqhoang** [comment]
Hey @rithvikb23, you can templatize plans by changing any variable reference to reference the root object, e.g., `@root.<variable_name>`

In your example, Step 1 would be changed to:
```
{{!-- Step 1: Setting the Key Values --}}
{{set "ticker" @root.ticker}}

{{!-- Step 2: Calling the Custom Helper --}}
{{set "response" ((CurrentPricePlugin-getCurrentPrice ticker=ticker}}

{{!-- Step 2: Output the Result --}}
{{json response}}
```

Then you can invoke the plan with a set of arguments you want to use:
```
var initialArguments = new KernelArguments()
        {
            { "ticker", "NVDA" }
        };

var result = await plan.InvokeAsync(kernel, initialArguments);
```

Likewise, you can call planner.CreatePlanAsync with a KernelArguments parameter.

**Comment by teresaqhoang** [comment]
The changes are now available as part of release dotnet-1.5.0 https://github.com/microsoft/semantic-kernel/releases/tag/dotnet-1.5.0

Will close this as resolved, feel free to re-open if you're still seeing the issue persist consistently.
